data/scripts/dependency-resolver.py

```diff
@@ -144,40 +205,125 @@ def handle_python_check(args, conf):
 
     cmd = "%s %s" % (sys.executable, f.name)
     output, status = run_command(cmd)
-    makefile_vars["HAVE_PYTHON_%s" % dep] = "y" if status else "n"
+    context.add_cond_makefile_var("HAVE_PYTHON_%s" % dep, "y" if status else "n")
+
+def handle_cflags_check(args, conf, context):
+    check_cflags = conf.get("cflags")
+    append_to = conf.get("append_to")
+    source = cstub.format(headers="", fragment="")
+
+    if not append_to or not check_cflags:
+        return
+
+    wanted = []
+    for i in check_cflags:
+        wanted.append(i)
+
+    flags = " ".join(wanted)
+    success = compile_test(source, args.compiler, "-Werror %s" % flags, None)
+    if success:
+        context.add_append_makefile_var(append_to, flags)
+        return
+
+    supported = []
+    for i in check_cflags:
+        # must acumulate the tested one so we handle dependent flags like -Wformat*
+        flags = "%s %s" % (" ".join(supported), i)
+        success = compile_test(source, args.compiler, "-Werror %s" % flags, None)
+        if success:
+            supported.append(i)
+
+    if supported:
+        context.add_append_makefile_var(append_to, " ".join(supported))
 
 type_handlers = {
     "pkg-config": handle_pkgconfig_check,
     "ccode": handle_ccode_check,
     "exec": handle_exec_check,
     "python": handle_python_check,
+    "cflags": handle_cflags_check,
 }
 
-def var_str(items):
+def format_makefile_var(items):
     output = ""
     for k,v in items:
-        if not v: continue
-        output += "%s ?= %s\n" % (k, v)
+        if not v or not v["value"]: continue
+        output += "%s %s %s\n" % (k, v["attrib"], v["value"])
     return output
 
-def makefile_gen(args):
-    output = ""
-    output += var_str(makefile_vars.items())
-    output += var_str(cfg_cflags.items())
-    output += var_str(cfg_ldflags.items())
+def makefile_gen(args, context):
+    output = format_makefile_var(context.get_makefile_vars().items())
     f = open(args.makefile_output, "w+")
     f.write(output)
     f.close()
 
-def kconfig_gen(args):
+def kconfig_gen(args, context):
     output = ""
-    for k,v in cfg_kconfig.items():
-        output += "config {config}\n{indent}bool\n{indent}default {enabled}\n". \
-                  format(config=k, indent="       ", enabled=v)
+    for k,v in context.get_kconfig().items():
+        output += "config {config}\n{indent}{ktype}\n{indent}default {enabled}\n". \
+                  format(config=k, indent="       ", ktype=v["type"], enabled=v["value"])
     f = open(args.kconfig_output, "w+")
     f.write(output)
     f.close()
 
+def run(args, dep_checks, context):
+    for i in dep_checks:
+        handler = type_handlers.get(i["type"])
+        if not handler:
+            print("Could not handle type: %s" % i["type"])
+            continue
```

Comment:
It's a programmer error: it's better to abort early and break the build during development rather than just print something that can be overlooked and move on.